### ISSUE TITLE
Experimental Tiles builds crash badly when trying to launch outside of the correct working directory

### ISSUE THREAD

**Issue by narc0tiq** [CleverRaven/Cataclysm-DDA#25331]
**Describe the bug**  
There is no useful error message when Cataclysm fails to start from outside its normal correct working directory. We should be checking for this condition and fail early with a useful message (perhaps mentioning the `cataclysm-launcher` script).

**To Reproduce**  
1. Unpack a Linux experimental Tiles build (e.g., one of [http://dev.narc.ro/cataclysm/jenkins-latest/Linux_x64/Tiles/](these)).
2. Change directory to somewhere else (e.g., `cd ~`)
3. Try to launch `/path/to/cataclysm-tiles`

**Expected behavior**  
Cataclysm should determine that the current working directory does not contain the `data` folder (or other core folders like `gfx`, `lang`, etc.) and return with a non-zero exit code and a useful error message.

For example: 
```
Fatal: can't find `data` directory. Please ensure the current working directory is
correct (perhaps you meant to start `cataclysm-launcher`?).
```

**Versions and configuration:**
 - OS: Linux_x64, but probably applicable to all platforms (this shouldn't be platform-specific)
 - Game Version: likely 47298ed1dea10582ae57a3decfe548640ac576de
 - Graphics version: Tiles; may also apply to Curses
 - Mods loaded: N/A

**Additional context**  
Found this from <https://old.reddit.com/r/cataclysmdda/comments/9cgd65/a_tip_for_anyone_on_ubuntu_and_maybe_other/>, where the OP had a lot of trouble finding out that `cataclysm-launcher` was what he really wanted. We should be helping reduce friction for these cases.

**Comment by maniak1349** [comment]
Can confirm for Windows x86 (Windows, tiles, 0.C-29661-g765578f).

Shouldn't cataclysm look for it's game data (but not for saves and config) in the directory where it's binary is like most of standalone/portable applications do?

**Comment by cainiaowu** [comment]
can confirm when building with VS2015 debug setting(before changing working directory), crash happens somewhere inside load_all_from_json.

**Comment by pingpong2011** [comment]
For safety reasons, Windows prevents loading/modifying from the local folder if it's in Program Files and you're not in admin mode. On Linux, a comparable situation would be not running from a /home/user/.*/ directory versus running from a /usr/bin/. This would only matter if the game did a "full install". Default paths should be added to both systems (%APPDATA% on windows, ~ or $HOME for most sh's) for anyone going so far.

**Comment by kevingranade** [comment]
Fixed by #28148
